app/hero.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+
+class Hero:
+    def __init__(
+            self,
+            name: str = "Default knight",
+            power: int = 20,
+            hp: int = 50,
+            armor: list = None,
+            weapon: dict = None,
+            potion: dict = None
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armor = armor if armor else []
+        self.weapon = weapon if weapon else {}
+        self.potion = potion if potion else {}
+        self.protection = 0
+
+    @classmethod
+    def create_from_config(cls, config: dict) -> Hero:
+        return cls(
+            name=config.get("name"),
+            power=config.get("power"),
+            hp=config.get("hp"),
+            armor=config.get("armour"),
+            weapon=config.get("weapon"),
+            potion=config.get("potion")
+        )
+
+    def put_on_armor(self) -> None:
+        if self.armor:
+            for item in self.armor:
+                self.protection += item["protection"]
```

Comment:
you can remove if statement because armor is a list with data or empty list(as you set it in init), thus it'll loop through all armour or not because it's empty